src/citrine/builders/design_spaces.py

```diff
@@ -1,40 +1,59 @@
 import pandas as pd
 from itertools import product
-from typing import Mapping, Sequence, Any
+from typing import Mapping, Sequence, Any, List
 
+from citrine.informatics.design_spaces import EnumeratedDesignSpace
+from citrine.informatics.descriptors import Descriptor
 
 def enumerate_cartesian_product(
-    dof_grids: Mapping[str, Sequence],
-) -> Sequence[Mapping[str, Any]]:
-    '''[ALPHA] Given a dict of lists or tuples of values for each degree of freedom,
+    design_grids: Mapping[str, Sequence],
+    descriptors: List[Descriptor],
+    name: str = 'Enumerated Cartesian product design space',
+    description: str = '',
+) -> EnumeratedDesignSpace:
```

Comment:
@maxhutch I'm making `name` and `description` kwargs with defaults to keep these functions lightweight. `descriptors` however is required - but I do not perform any kind of checks to ensure the descriptors are correct.